public/js/pieces/algobots.js

```diff
@@ -0,0 +1,2728 @@
+          j: [.2, .3, .1, .35, .6, .15, .5, .25],
+          b: ["isolation:isolate"],
+          n: [230.13, 568.3, 400, 400.19, 400.45, 400.09, 397.83, 397.92, 400.54, 399.4, 277.72, 524.19, 306.55, 493.45, 277.54, 523.54, 243.28, 556.72, 242.98, 556.27, 257.83, 542.17, 330.32, 469.68, 317.71, 457.07, 318.95, 469.78, 458.41, 324.81, 324.32, 400.04, 475.18, 474.69, 301.9, 498.1, 355.62, 444.38, 399.46, 399.25, 399.98, 399.97, 422.24, 377.75],
+          o: [328.81, 597.73, 244.31, 597.72, 245.3, 443.73, 533.57, 527.45, 271.9, 340.3, 305.42, 283.17, 246.24, 238.89, 243.86, 236.51, 488.58, 501.86, 490.94, 477.12, 488.09, 354.65, 356.52, 339.33, 317.85, 319.71, 302.53, 353.65, 497.81, 492.97, 490.2, 496.12, 491.28, 331.99, 334.81, 329.47, 308.88, 496.23, 474.86, 505.76, 472.67, 469.67, 467.78, 504.9, 501.9, 500.02],
+          l: [6.45, 8.62, 156.72, 30.08, 187.16, 129.64, 106.56, 5.59, 19.53, 5.11, 15.15, 22.86, 4.93, 2.41],
+          m: [13.51, 10.63, 6.47, 4.58, 11.47, 2.56, 11.2, 16.9, 1.78],
+          x: [217.55, 563.05, 417.31, 372.83],
+          y: [314.9, 470.32, 501.18],
+          q: [17.25, 9.86],
+          p: [32.03, 21.81],
+          t: ["rotate(-180 571.675 330.915)", "rotate(-80.781 243.28447919 246.2363546)", "rotate(-180 377.755 512.08)"],
+          r: [5.17, 16.37, 2.89, 40.26, 19.47, 14.55, 7.79, 42.86, 18.19, 31.93, 34, 16.82, 21.82, 7.41, 30.51, 48.17, 51.29, 24.16, 32.03, 36.45, 24.09, 27.41, 1.83, 7.12]
+     },
+     // Embed common attributes to shorten file
+     d = {
```

Comment:
This seems to be a clever way to save space in the file. Rather than explicitly typing each shape, the author just calls d[1] whenever they want a line representation